test/signer_test.go

```diff
@@ -0,0 +1,196 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/Azure/dalec"
+	"github.com/moby/buildkit/client/llb"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+)
+
+// TestHandlerTargetForwarding tests that targets are forwarded to the correct frontend.
+// We do this by registering a phony frontend and then forwarding a target to it and checking the outputs.
+func TestSignerForwarding(t *testing.T) {
+	runTest := func(t *testing.T, f gwclient.BuildFunc) {
+		t.Helper()
+		ctx := startTestSpan(baseCtx, t)
+		testEnv.RunTest(ctx, t, f)
+	}
+
+	t.Run("test mariner2 signing", func(t *testing.T) {
```

Comment:
I think all this should move to the distro-level testing since its really making sure the distro implments signing correctly.

Reply:
Seems like we'd need to support `PackageConfig` at the root of the spec like we do other target fields.

Reply:
> I think all this should move to the distro-level testing since its really making sure the distro implments signing correctly.

this part is done. Still need to support `PackageConfig` at spec root.

Reply:
Set up PackageConfig at the root level, so both parts of this are done now.